src/sequence_middleware.ts

```diff
@@ -0,0 +1,42 @@
+import * as TE from "fp-ts/lib/TaskEither";
+import * as E from "fp-ts/lib/Either";
+import * as T from "fp-ts/lib/Task";
+import * as RA from "fp-ts/lib/ReadonlyArray";
+import { pipe } from "fp-ts/lib/function";
+import * as express from "express";
+import { IRequestMiddleware } from "./request_middleware";
+import { IResponse } from "./responses";
+
+export type SequenceMiddlewareT = <E>(
+  defaultError: IResponse<E>
+) => <R0, R1>(
+  middleware0: IRequestMiddleware<unknown, R0>,
+  middleware1: IRequestMiddleware<unknown, R1>
+) => IRequestMiddleware<E, R0 | R1>;
+
+export const SequenceMiddleware: SequenceMiddlewareT = defaultError => (
+  middleware0,
+  middleware1
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+) => async <R extends express.Request, R0, R1>(request: R) =>
+  await pipe(
+    [middleware0, middleware1],
+    RA.map(middleware =>
+      pipe(
+        TE.tryCatch(
+          async () =>
+            (await middleware(request)) as E.Either<
+              IResponse<unknown>,
+              R0 | R1
+            >,
```

Comment:
Be aware that this explicit casting is very dangerous as it may undermine the type-safeness of the whole function (and the application in worst cases).
I see it's not that easy to avoid the cast in this context so let's stick with it until we find a proper solution. However, I advocate for:
* Highlight the manual casting by writing a comment (so that interested readers will spot the `as` quickly)
* Try to stress result type in tests (I did a bit in https://github.com/pagopa/io-ts-commons/pull/406/commits/8290392eae9631f6a7140e4484995b5807dec5e2)